experiment_api/experiment.py

```diff
@@ -0,0 +1,252 @@
+        for i in range(0, self.num_cams):
+            output.append(self.__get_camset())
+        self.__unset_lists()
+        return output
+
+class Image(object):
+    """ Assume image's name is in the format of carId_cameraId_timestamp_binary.jpg
+    """
+    def __init__(self, img_name, img_dir, img_type):
+        self.name = img_name
+        self.filepath = img_dir + "/" + img_name
+        self.type = img_type
+        self.__splitter = self.name.split("_")
+        self.car_id = self.__set_car_id()
+        self.camera_id = self.__set_camera_id()
+        self.timestamp = self.__set_timestamp()
+        self.binary = self.__set_binary()
+
+    def get_name(self):
+        return self.name
+
+    def __set_car_id(self):
+        car_id = int(self.__splitter[0])
+        return car_id
+
+    def get_car_id(self):
+        return self.car_id
+
+    def __set_camera_id(self):
+        camera_id = int(get_numeric(self.__splitter[1]))
+        return camera_id
+
+    def get_camera_id(self):
+        return self.camera_id
+
+    def __set_timestamp(self):
+        timestamp = datetime.datetime.fromtimestamp(int(self.__splitter[2]))
+        return timestamp
+
+    def get_timestamp(self):
+        return self.timestamp
+
+    def get_timestamp_in_str(self):
+        # Year-Month-Date Hour:Minute:Second
+        return self.timestamp.strftime("%Y-%m-%d %H:%M:%S")
+
+    def __set_binary(self):
+        binary = int(os.path.splitext(self.__splitter[3])[0])
+        return binary
+
+    def get_binary(self):
+        return binary
+
+    def __hash__(self):
+        return hash(self.name)
+
+    def __eq__(self, other):
+        return self.name == other.name
+
+# -----------------------------------------------------------------------------
+#  Helper Functions
+# -----------------------------------------------------------------------------
+
+def get_numeric(string):
+    """ Extract the numeric value in a string.
+    Args:
+        string
+    Returns:
+        a string with only the numeric value extracted
+    """
+    return re.sub('[^0-9]','', string)
+
+def should_drop(drop_percentage):
+    """ Based on the given percentage, provide an answer 
+    whether or not to drop the image.
+    Args:
+        drop_percentage: the likelihood of a drop 
+    Returns:
+        a boolean whether to drop or not drop the image
+    """
+    return random.randrange(100) < drop_percentage
+
+def select_cameras(num_cams):
+    """ Select camera ids based on the number of cameras specified.
+    Args:
+        num_cams: number of cameras 
+    Returns:
+        a list of camera ids
+    """
+    return random.sample(range(1, 21), num_cams)
+
+# -----------------------------------------------------------------------------
+#  Execution
+# -----------------------------------------------------------------------------
+
+def main():
+    # inputs
+    veri_unzipped_path = ""
+    num_cams = 5
+    num_cars_per_cam = 3
+    drop_percentage = 30
```

Comment:
Can we make `drop_percentage` a float ranging from [0, 1]? This allows more fine-grained percentages.